### PR TITLE
Improve performance of fileio._CompressedFile

### PR DESCRIPTION
**Pull request by charlesccychen**
This change fixes the timing out of the "test_read_entire_file_bzip2_large" test.

### CODE REVIEW

**Comment by charlesccychen** [comment]
R: @chamikaramj

File: sdks/python/apache_beam/io/fileio.py [hunk 1]
```diff
@@ -664,6 +664,7 @@ def __init__(self,
         self._decompressor = bz2.BZ2Decompressor()
       else:
         self._decompressor = zlib.decompressobj(self._gzip_mask)
+      self.read_eof = False
```
Comment (chamikaramj) [inline sdks/python/apache_beam/io/fileio.py]: self._read_eof ?

**Comment by chamikaramj** [comment]
LGTM

**Comment by charlesccychen** [comment]
Thanks!

R: @robertwb for merge

File: sdks/python/apache_beam/io/fileio.py [hunk 2]
```diff
@@ -728,10 +729,14 @@ def _fetch_to_internal_buffer(self, num_bytes):
...
         else:
           self._data += self._decompressor.flush()
+        # Record that we have hit the end of file, so we won't unnecessarily
+        # repeat the completeness verification step above.
+        self._read_eof = True
         return
 
   def _read_from_internal_buffer(self, num_bytes):
     """Read up to num_bytes from the internal buffer."""
+    # TODO: this can be optimized to avoid a string copy operation.
```
Comment (dhalperi) [inline sdks/python/apache_beam/io/fileio.py]: driveby that many of these TODOs make good JIRA issues, especially to empower new potential contributors to the SDK.

**Comment by robertwb** [comment]
This can now be closed.

**Comment by charlesccychen** [comment]
Thanks!
